### PR TITLE
GCP Vertex AI LLM Integration

### PR DESCRIPTION
**Pull request by ishleenk17**
GCP Vertex AI LLM Integration

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates https://github.com/elastic/obs-integration-team/issues/64

## Screenshots
gcp-vertexai-overview-dashboard

### CODE REVIEW

**Review by muthu-mps — APPROVED** [review]
LGTM!

**Comment by agithomas** [comment]
Based on the screenshot attached to the issue, here below are some of the suggestions for improvements

1. You may consider removing the decimals from Total invocations
2. Consider including the account selection / region control dropdown  at the top of the dashboard
3. Very surprised to see the model invocation latency dashboard is empty. For every LLM request , chance of not having an LLM invocation latency is nil. Please re-look if the dashboard is correctly configured 
4. Are there any available attributes that identify which of the CPU's are utilised? Please check the unit mapping to select percentage (if that is the case so). Same with memory as well. Are there any available attributes to identify the resource's whose memory is referred in the visualisation
5. Network utilisation graph has label "Received Bytes". But, it is indented for Sent & received bytes. Suggest to change the Y-axis label to network utilisation. 
6. Observed `[GCP Billing]` prefix towards the end of the screenshot. Consider removing the prefix.
7.

**Comment by agithomas** [comment]
> Based on the screenshot attached to the issue, here below are some of the suggestions for improvements
> 
> 1. You may consider removing the decimals from Total invocations
> 2. Consider including the account selection / region control dropdown  at the top of the dashboard
> 3. Very surprised to see the model invocation latency dashboard is empty. For every LLM request , chance of not having an LLM invocation latency is nil. Please re-look if the dashboard is correctly configured
> 4. Are there any available attributes that identify which of the CPU's are utilised? Please check the unit mapping to select percentage (if that is the case so). Same with memory as well. Are there any available attributes to identify the resource's whose memory is referred in the visualisation
> 5. Network utilisation graph has label "Received Bytes". But, it is indented for Sent & received bytes. Suggest to change the Y-axis label to network utilisation.
> 6. Observed `[GCP Billing]` prefix towards the end of the screenshot. Consider removing the prefix.

I see that the screenshot attached to the PR is not updated. So, some of the comments mentioned above may be obsolete and may not need any further action.

**Comment by agithomas** [comment]
1. Please do consider having received & sent as a single panel. Having them separate may take away the advantage of in-line visualisation comparison. 
2. Any reason why 50 percentile latency is considered for determining LLM invocation latency?

File: packages/gcp_vertexai/_dev/build/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# GCP Vertex AI
+
+## Overview
+
+Vertex AI is a platform that enables the training and deployment of machine learning models and AI applications. It aims to streamline and expedite the development and deployment process for ML models, offering a variety of features and integrations tailored for enterprise-level workflows.
```
Comment (agithomas) [inline packages/gcp_vertexai/_dev/build/docs/README.md:5]: Would it be good to give a link to the Google's Vertex AI page from the documentation?

File: packages/gcp_vertexai/changelog.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# newer versions go on top
+- version: "0.0.1"
+  changes:
+    - description: Initial draft of the GCP Vertex AI package
```
Comment (agithomas) [inline packages/gcp_vertexai/changelog.yml:4]: Not sure if `draft` is the most commonly used term. `build` or `release` may be more appropriate,i think.

File: packages/gcp_vertexai/data_stream/metrics/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
+---
+description: Pipeline for parsing GCP Vertex AI metrics.
```
Comment (agithomas) [inline packages/gcp_vertexai/data_stream/metrics/elasticsearch/ingest_pipeline/default.yml:2]: We have used available semconv metric names for AWS bedrock field names.

Is the below field name pattern the preferred pattern for GCP? 

cc @lalit-satapathy

**Comment by agithomas** [comment]
Is there any reason why the TSDS enablement is not present?

**Comment by ishleenk17** [comment]
> Is there any reason why the TSDS enablement is not present?

We are doing this for Tech Preview and to start getting the initial feedbacks. 
Will be enabled when we move it to Beta.

**Pull request merged**